File: ruby/1.rb
# If we list all the natural numbers below 10 that are multiples 
# of 3 or 5, we get 3, 5, 6 and 9. The sum of these multiples is 23.

# Find the sum of all the multiples of 3 or 5 below 1000.

def sumOfMultsOf3and5(max)
  sum = 0
  for i in 0..max-1
    if i % 3 == 0 || i % 5 == 0
      sum += i
    end
  end
  sum
end

puts sumOfMultsOf3and5(1000)

File: rust/src/1.rs
// Sum of the multiples of 3 and 5 below 1000
fn main() {
    let max = 1000;
    let total = sum(max - 1, 0);
        

    println!("total: {}", total)
}

fn sum(remaining: u32, total: u32) -> u32 {
    if remaining == 0 {
        return total;
    }

    if remaining % 3 == 0 || remaining % 5 == 0 {
        return sum(remaining - 1, total + remaining);
    } else {
        return sum(remaining - 1, total);
    }
}
